plugins/resource_monitor_plugin/test/test_resmon_plugin.cpp

```diff
@@ -0,0 +1,144 @@
+#define BOOST_TEST_MODULE test_resmom_plugin
+#include <boost/test/included/unit_test.hpp>
+
+#include <fc/variant_object.hpp>
+
+#include <eosio/resource_monitor_plugin/resource_monitor_plugin.hpp>
+
+using namespace eosio;
+//using namespace eosio::resource_monitor;
+using namespace boost::system;
+
+namespace bfs = boost::filesystem;
+
+// For program options
+namespace bpo = boost::program_options;
+using bpo::options_description;
+using bpo::variables_map;
+
+struct resmon_fixture {
+   void set_program_options() {
+      options_description dummy;
+      _my.set_program_options(dummy, _cfg);
+   }
+
+   void initialize(const std::vector<std::string>& args){
+      // We only have at most 3 arguments. OK to hardcodied in test
+      // programs.
+      const char* argv[10];
+      // argv[0] is program name, no need to fill in
```

Comment:
You should have some kind of assert here that the argument size is what you expect.  If not then it could silently overflow argv if in the future if a lot more args are added to resmon.

Reply:
Thanks for pointing out this! Sorry for the overlook. Will change.